docs/aws/deploy-s3.md

```diff
@@ -0,0 +1,52 @@
+# Setup an Application using Simple Storage Service (S3)
+This document explains how to setup an application that does not need a server to be running i.e. is a static application, for e.g. static websites.
+
+## Services Used
+- S3
+- CloudFront
+- Route 53
+
+## Steps
+
+### Setting up the S3 Bucket
+1. __Role__: Switch to the appropriate role in which the content for the application needs to exist in.
+2. __Create Bucket__: Click on the "Create Bucket" button in S3, give your bucket a name and select "US Standard" as the region.
+3. __Upload Content__: Click on "Upload" and drag and drop your files into the bucket.
+4. __Setup Hosting__: Next you need to setup hosting for the application. Here's how you do it:
+  1. Select the "Properties" tab and expand the "Static Website Hosting" section.
+  2. Select "Enable website hosting" and type in your application entry-point (Index Document) and the default error page you would like to show (set it to the same value of the Index Document if you don't have a custom error page).
+  3. Click "Save". You should now see an "Endpoint" url that you can access your bucket's contents at. However, this url won't be accessible until you setup permissions for the bucket as described in the next step.
+5. __Permissions__: Adequate permissions need to be set on the bucket to enable access to it. The steps for this are:
+  1. Expand the "Permissions" section in the "Properties" tab.
+  2. Select "Add Bucket Policy" and click on the "AWS Policy Generator" link in the bottom-left corner. This will open a new window/tab with the Policy Generator page.
+  3. In the Policy Generator page:
+    1. Select the "Policy Type" as "S3 Bucket Policy". 
+    2. Set the "Principal" to "*"(without quotes) to allow access to all the content in the bucket. 
+    3. For the "Action", you will need to at least select the "GetObject" permission. If you want to allow uploads to the S3 bucket from anywhere outside AWS, you should also enable the "PutObject" permission.
```

Comment:
noooooo never allow putObject with "*" as the principal, since this lets anybody upload files. 

For uploading outside of AWS you'd create a new IAM user with a policy that allows access to the S3 bucket with the putObject action.